src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java

```diff
@@ -136,9 +137,11 @@ private static String prepareHistoricalFifoReadEBDString(
    * @return dataPoints
    * @throws IOException for parsing Exceptions
    * @throws JSONException for JSON parsing Exceptions
+   * @throws TimeoutException timeout for reading event file
+   * @throws CircularizedFileException when event log indicates circularized event
    */
   private static ArrayList parseEBDHistoricalLogExportResponse(Exporter exporter)
-      throws IOException, JSONException {
+      throws IOException, JSONException, TimeoutException, CircularizedFileException {
```

Comment:
See earlier comment about TimeoutException being documented as Java 1.5 and newer.

Reply:
changed to EBdTimeout custom class